crates/bevy_math/src/common_traits.rs

```diff
@@ -47,29 +50,99 @@ pub trait VectorSpace:
     /// of the interval `[0,1]` is allowed.
     #[inline]
-    fn lerp(self, rhs: Self, t: f32) -> Self {
-        self * (1. - t) + rhs * t
+    fn lerp(self, rhs: Self, t: Self::Scalar) -> Self {
+        self * (Self::Scalar::ONE - t) + rhs * t
     }
 }
 
 impl VectorSpace for Vec4 {
+    type Scalar = f32;
     const ZERO: Self = Vec4::ZERO;
 }
 
 impl VectorSpace for Vec3 {
+    type Scalar = f32;
     const ZERO: Self = Vec3::ZERO;
 }
 
 impl VectorSpace for Vec3A {
+    type Scalar = f32;
     const ZERO: Self = Vec3A::ZERO;
 }
 
 impl VectorSpace for Vec2 {
+    type Scalar = f32;
     const ZERO: Self = Vec2::ZERO;
 }
 
-impl VectorSpace for f32 {
+impl VectorSpace for DVec4 {
+    type Scalar = f64;
+    const ZERO: Self = DVec4::ZERO;
+}
+
+impl VectorSpace for DVec3 {
+    type Scalar = f64;
+    const ZERO: Self = DVec3::ZERO;
+}
+
+impl VectorSpace for DVec2 {
+    type Scalar = f64;
+    const ZERO: Self = DVec2::ZERO;
+}
+
+// Every scalar field is a 1-dimensional vector space over itself.
+impl<T: ScalarField> VectorSpace for T {
+    type Scalar = Self;
+    const ZERO: Self = Self::ZERO;
+}
+
+/// A type that supports the operations of a scalar field. An implementation should support:
+/// - Addition and subtraction
+/// - Multiplication and division
+/// - Negation
+/// - Zero (additive identity)
+/// - One (multiplicative identity)
+///
+/// Within the limitations of floating point arithmetic, all the following are required to hold:
+/// - (Associativity of addition) For all `u, v, w: Self`, `(u + v) + w == u + (v + w)`.
+/// - (Commutativity of addition) For all `u, v: Self`, `u + v == v + u`.
+/// - (Additive identity) For all `v: Self`, `v + Self::ZERO == v`.
+/// - (Additive inverse) For all `v: Self`, `v - v == v + (-v) == Self::ZERO`.
+/// - (Associativity of multiplication) For all `u, v, w: Self`, `(u * v) * w == u * (v * w)`.
+/// - (Commutativity of multiplication) For all `u, v: Self`, `u * v == v * u`.
+/// - (Multiplicative identity) For all `v: Self`, `v * Self::ONE == v`.
+/// - (Multiplicative inverse) For all `v: Self`, `v / v == v * v.inverse() == Self::ONE`.
+/// - (Distributivity over addition) For all `a, b: Self`, `u, v: Self`, `(u + v) * a == u * a + v * a`.
+pub trait ScalarField:
+    Mul<Self, Output = Self>
+    + Div<Self, Output = Self>
+    + Add<Self, Output = Self>
+    + Sub<Self, Output = Self>
+    + Neg<Output = Self>
+    + Default
+    + Debug
+    + Clone
+    + Copy
+{
+    /// The additive identity.
+    const ZERO: Self;
+    /// The multiplicative identity.
+    const ONE: Self;
+
+    /// The multiplicative inverse of this element. This is equivalent to `1.0 / self`.
+    fn recip(self) -> Self {
+        Self::ONE / self
+    }
+}
+
+impl ScalarField for f32 {
     const ZERO: Self = 0.0;
+    const ONE: Self = 1.0;
+}
+
+impl ScalarField for f64 {
+    const ZERO: Self = 0.0;
+    const ONE: Self = 1.0;
```

Comment:
Very happy for this, I had to have my own extension trait in [`bevy_math_extensions`](https://github.com/Jondolf/bevy_math_extensions) just for the `ZERO` and `ONE` constants for `f64` :P